request.js

```diff
@@ -137,6 +137,7 @@ function Request (options) {
   self._redirect = new Redirect(self)
   self._tunnel = new Tunnel(self)
   self.init(options)
+  self.options = options
```

Comment:
This has a bit of a bad code smell because we already set lots of properties on `self` and want to be careful about adding more without good naming and clear reasons.